tests/model/DataObjectTest.php

```diff
@@ -564,50 +564,58 @@ public function testChangedFields() {
 		$newObj = new DataObjectTest_Player();
 		$newObj->FirstName = "New Player";
 		$this->assertEquals(
-			$newObj->getChangedFields(false, 2),
 			array(
 				'FirstName' => array(
 					'before' => null,
 					'after' => 'New Player',
-					'level' => 2
+					'level' => DataObject::CHANGE_VALUE
 				)
 			),
+			$newObj->getChangedFields(true, DataObject::CHANGE_VALUE),
 			'Initialised fields are correctly detected as full changes'
 		);
 	}
 
 	public function testIsChanged() {
 		$obj = $this->objFromFixture('DataObjectTest_Player', 'captain1');
+		$obj->NonDBField = 'bob';
```

Comment:
I think we should test that changes to composite fields are still detected as `isChanged()`.